pydantic/types.py

```diff
@@ -107,6 +121,60 @@ def validate(cls, value):
         return validate_email(value)[1]
 
 
+class UrlStr(str):
+    strip_whitespace = True
+    min_length = 1
+    max_length = 2 ** 16
+    default_schemes = {'http', 'https', 'ftp', 'ftps'}
+    schemes: Optional[Set[str]] = None
```

Comment:
I don't think we need `default_schemes`, just

```py
schemes = cls.schemes or {'http', 'https', 'ftp', 'ftps'}
```
below

Reply:
or just use `{'http', 'https', 'ftp', 'ftps'}` as the default here and below in `urlstr`.

(using a global variable for `DEFAULT_SCHEMAS` so the actual set isn't duplicated)

Reply:
fixed